### ISSUE TITLE
UpdateEntriesAsync raises error when issued in a batch

### ISSUE THREAD

**Issue by HEBOS** [object/Simple.OData.Client#375]
I am trying to issue update in a batch (avoiding many single patch http requests), I am using following code:

```
var client = new ODataClient(settings);
                var batch = new ODataBatch(client);
                IEnumerable<CachedMessage> updateResults;

                batch += async c => updateResults = await c
                    .For<CachedMessage>()
                        .Filter(x => !x.isSynced)
                        .Set(new Dictionary<string, object> { { "isSynced", true } })
                        .UpdateEntriesAsync(true);
                
                // Perform batch update
                await batch.ExecuteAsync();
```
				
However, I get an exception saying "An item with the same key has already been added.".
If I try to perform update without batch, it works as expected.

I narrowed down the code that causes the error, but I cannot find the reason:
Here is the code from **BatchWriterBase.cs**:
```
public async Task<ODataRequest> CreateBatchRequestAsync(
            IODataClient client, IList<Func<IODataClient, Task>> actions, IList<int> responseIndexes)
        {
            // Write batch operations into a batch content
            var lastOperationId = 0;
            foreach (var action in actions)
            {
                await action(client);      // < ------  THIS LINE IS CAUSING THE EXCEPTION
```

Here is the log:
		exception.StackTrace.ToString()	"   at Microsoft.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   
		at Microsoft.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccess(Task task)\r\n   
		at Microsoft.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()\r\n   
		at Simple.OData.Client.ODataClient.<ExecuteBatchAsync>d__114.MoveNext() in **ODataClient.Async.cs:line 1209**
		\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   
		at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   
		at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   
		at System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()\r\n   
		at WinSample.Form1.<button1_Click>d__1.MoveNext() in C:\\t\\Simple.OData.Client-master\\SampleWin\\WinSample\\Form1.cs:line 84"

**Comment by object** [comment]
Hi,

I will need to investigate this more but perhaps the problem is realated to the fact that UpdateEntries and DeleteEntries don't correspond to a single native OData operation - these are compound operations that internally make individual OData PUT/PATCH/DELETE calls, and when using from a batch they fail. If so, they may be incompatible with OData batch, single UpdateEntry and DeleteEntry will have to be used instead. But I will investigate more.

**Comment by object** [comment]
Fixed in version 4.26.0. Please try it out.
